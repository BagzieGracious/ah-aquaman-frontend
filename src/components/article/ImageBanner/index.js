import React, { Component } from "react";
import Banner from "../../../assets/img/banner.jpg";

export default class ImageBanner extends Component {
  render() {
    return (
      <React.Fragment>
        <section className="image-banner">
          <img src={Banner} alt="" />
          <div className="container">
            <div className="banner-content text-center">
              <h2>{this.props.info}</h2>
              <p>
                Home 
                {" "}
                <i className="fa fa-angle-right" /> 
                {" "}
                {this.props.page}
              </p>
            </div>
          </div>
        </section>
      </React.Fragment>
    );
  }
}
