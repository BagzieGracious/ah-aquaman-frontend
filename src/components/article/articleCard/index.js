import React, { Component } from "react";
import { Link } from "react-router-dom";
import parseDate from "../../../commons/getArticleDate";
import Ratings from "../ratingsComponent";

export class ArticleCard extends Component {
  render() {
    return (
      <article>
        <div className="article-img">
          <img className="img-fluid" src={this.props.image} alt="" />
        </div>

        <div className="article-text">
          <div className="inner-text">
            <Link to={`/article/${this.props.slug}`} className="article-title">
              <h4>{this.props.title}</h4>
            </Link>
            <p>
              {(this.props.body).slice(0, 205)}
            </p>
            <div className="date">
              <table className="table mb-0 borderless">
                <tbody>
                  <tr>
                    <td className="p-0">
                      <i className="fa fa-calendar" />
                      {" "}
                      {parseDate(this.props.created_at)}
                    </td>
                    <td className="p-0 text-center">
                      <i className="fa fa-clock" /> 
                      {" "}
                      {this.props.read_time}
                    </td>
                    <td className="p-0 text-center">
                      <Ratings rating={parseFloat(this.props.user_rating)} />
                    </td>
                    <td className="text-right p-0" onClick={() => document.location.href = `/profile/${this.props.author.username}`}>
                      {this.props.author.username}
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </article>
    );
  }
}

export default ArticleCard;
