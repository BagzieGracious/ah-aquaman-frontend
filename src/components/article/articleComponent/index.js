import React from "react";
import { Link } from "react-router-dom";
import Parser from "html-react-parser";
import EditButton from "../editArticleButton";
import DeleteButton from "../deleteArticleButton";
import RateButton from "../rateArticleButton";
import parseDate from "../../../commons/getArticleDate";
import TagComponent from "./tagComponent";
import CommentView from "../../../views/commentView";
import Ratings from "../ratingsComponent";
import RatingModal from "../ratingsModel";


function getImage(image) {
  if (image !== null) return "https://res.cloudinary.com/wasibani/image/upload/v1555329619/AuthorsHaven/background-image-3.jpg";
  return image;
}

const ArticleComponent = props => (

  <div className="col-lg-8">
    <div className="article-container">
      <div className="details">
        <img className="img-responsive avatar" src={getImage(props.article.image)} alt="author avatar" />
        <h4>{props.article.title}</h4>
        <Ratings rating={parseFloat(props.article.user_rating)} />
        <div className="user-details">
          <div className="float-left">

            {
              props.article.tagList.length > 0
                ? (props.article.tagList.map(tag => (<TagComponent tag={tag} key={tag} />)))
                : <Link to="#">No Tags</Link>
            }
          </div>
          <div className="float-right">
            <div className="media">
              <div className="media-body">
                <h5>{props.article.author.username}</h5>
                <p>
                  { parseDate(props.article.created_at) }
                  <span className="ml-2">
                    <i className="fa fa-clock" /> 
                    {props.article.read_time}
                  </span> 
                </p>
              </div>
            </div>
          </div>
        </div>

        <p>
          {Parser(String(props.article.body))}
        </p>
        <div className="share">
          <h1 className="articleShare">You can share this article here</h1>
          <a data-tip="Share on FB" rel="noopener noreferrer" target="_blank" href={props.article.shareLinks.facebookShare}><span className="m-2"><i className="fab fa-facebook-square  fa-lg" /></span></a>
          <a data-tip="Share on twitter" rel="noopener noreferrer" target="_blank" href={props.article.shareLinks.twitterShare}><span className="m-2"><i className="fab fa-twitter-square  fa-lg" /></span></a>
          <a data-tip="Share on email" rel="noopener noreferrer" target="_blank" href={props.article.shareLinks.mailshare}><span className="m-2"><i className="fas fa-envelope-square  fa-lg" /></span></a>
        </div>
        <EditButton
          onClick={props.onClick}
          slug={props.slug}
          username={props.article.author.username}
        />

        <DeleteButton
          onClick={props.onClick}
          slug={props.slug}
          username={props.article.author.username}
        />

        <RateButton username={props.article.author.username} />
        <div className="separator" />
      </div>
      <div className="comments-container">
      <CommentView slug={props.slug}/>
      </div>
    </div>
    <RatingModal article={props.article} />
  </div>
);

export default ArticleComponent;
