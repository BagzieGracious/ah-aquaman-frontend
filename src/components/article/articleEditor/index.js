import React from "react";
import InputField from "../../../commons/inputField";
import Button from "../../../commons/buttons";

const Editor = ({
  onSubmit, title, description, body, onInputChange,
}) => (
  <div className="col-lg-8">
    <div className="article-container">
      <div className="details create-article">
        <h4>Publish Article</h4>
        <div className="separator" />

        <form onSubmit={onSubmit} className="mt-5">
          <div>
            <div className="form-group">
              <InputField
                name="title"
                className="form-control "
                type="text"
                placeholder="Title"
                value={title}
                onInputChange={onInputChange}
                maxLength="200"
              />
            </div>

            <div className=" form-group ">
              <InputField
                name="description"
                className="form-control "
                type="text"
                placeholder="What's this article about?"
                value={description}
                onInputChange={onInputChange}
                maxLength="200"
              />
            </div>
            <div className="form-group">
              <textarea className="form-control" value={body} onChange={onInputChange} placeholder="Tell your story...... " name="body" />
            </div>
            <div className="form-group">
              <Button 
                className="btn btn-article btn-primary float-left"
                id="publish"
                type="submit" 
                text="Publish Article"
              />
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
);

export default Editor;
