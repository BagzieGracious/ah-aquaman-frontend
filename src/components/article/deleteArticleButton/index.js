import React, { Component } from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import { toast } from "react-toastify";
import jwt from "jsonwebtoken";
import { DeleteArticle } from "../../../actions/articleActions/createArticle";


export class DeleteButton extends Component {
  constructor(props) {
    super(props);
    this.handleDelete = this.handleDelete.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.isArticleDeleted) {
      this.props.history.push("/");
      toast.success("Article successfully deleted", {
        position: toast.POSITION.TOP_CENTER,
        autoClose: 3000,
        hideProgressBar: false,
        pauseOnHover: true,
      });
    }
  }

  handleDelete(event) {
    event.preventDefault();
    const confirmDelete = window.confirm("Are you sure you want to delete?");
    if (confirmDelete) {
      this.props.DeleteArticle(this.props.slug);
    }
  }

  render() {
    const decoded = jwt.decode(localStorage.getItem("token"));
    if (decoded) {
      const loggedInUser = decoded.username;
      const hideButton = loggedInUser === this.props.username;
      return (
        <span
          className="btn btn-danger mr-3"
          slug={this.props.slug}
          onClick={this.handleDelete}
          role="button"
          tabIndex="0"
          onKeyPress={() => {}}
          hidden={!hideButton}
          data-tip="Delete the Article"
        >
          <i className="fas fa-trash-alt" />
          {" "}
          Delete
        </span>
      );
    }
    return null;
  }
}


export const mapStateToProps = state => ({
  isArticleDeleted: state.articleCreateReducer.isArticleDeleted,
});

export default withRouter(connect(mapStateToProps, { DeleteArticle })(DeleteButton));
