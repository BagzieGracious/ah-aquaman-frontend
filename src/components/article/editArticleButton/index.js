import React, { Component } from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import jwt from "jsonwebtoken";

export class EditButton extends Component {
  render() {
    const decoded = jwt.decode(localStorage.getItem("token"));
    if (decoded) {
      const loggedUser = decoded.username;
      const hideButton = loggedUser === this.props.username;
      return (
        <Link to={`/editor/${this.props.slug}`}>
          <span
            className="btn btn-primary mr-3"
            slug={this.props.slug}
            hidden={!hideButton}
            role="button"
            tabIndex="0"
            onKeyPress={() => {}}
            data-tip="Edit the Article"
          >
            <i className="fa fa-pencil-alt" />
            {" "}
            Edit
          </span>
        </Link>
      );
    }
    return null;
  }
}

export const mapStateToProps = state => ({
  EditState: state.articleCreateReducer,
});

export default connect(mapStateToProps)(EditButton);
