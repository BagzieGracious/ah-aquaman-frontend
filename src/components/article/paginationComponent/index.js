import React, { Component } from "react";
import Pagination from "react-js-pagination";

export default class PaginationComponent extends Component {
  state = {
    activePage: 1,
  };

  render() {
    return (
      <div className="col-sm-12 article-pagination-component">
        <Pagination
          activePage={this.state.activePage}
          itemsCountPerPage={6}
          totalItemsCount={100}
        />
      </div>
    );
  }
}
