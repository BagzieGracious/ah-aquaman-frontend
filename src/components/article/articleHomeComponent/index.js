import React, { Component } from "react";
import { connect } from "react-redux";
import { ArticleCard } from "../articleCard";
import fetchArticlesAction from "../../../actions/articleActions/fetchArticle";
import PaginationComponent from "../paginationComponent";

export class Articles extends Component {
  componentWillMount = () => {
    this.props.fetchArticlesAction();
  };

  render() {
    const { articles } = this.props;

    const articleItems = articles.map((article, index) => {
      if (!article.image) {
        article.image = `https://source.unsplash.com/random/${index}`;
      }
      return <ArticleCard key={article.slug} {...article} />;
    });

    console.log(articles);

    return (
      <div className="col-lg-8">
        <div className="article-container">{ articleItems }</div>
        <PaginationComponent />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  articles: state.fetchArticlesReducer.articles,
});

export default connect(
  mapStateToProps,
  { fetchArticlesAction },
)(Articles);
