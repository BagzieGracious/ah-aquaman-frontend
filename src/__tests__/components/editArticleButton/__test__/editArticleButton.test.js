import React from "react";
import { shallow } from "enzyme";
import { EditButton, mapStateToProps } from "../../../../components/article/editArticleButton";

const user = {
  username: "Roy",
  email: "roy@andela.com",
  password: "spidermansucks123",
};

const generateToken = (user) => {
  const jwt = require("jsonwebtoken");
  const token = jwt.sign(user, "sfhjjkdfjdkdfjdk", { expiresIn: 60 * 60 * 24 });
  return token; 
};

const token = generateToken(user);

describe("Edit button component test", () => {
  const props = {
    slug: "article-slug",
    hidden: "false",
  };

  beforeEach(() => {
    localStorage.setItem("token", token);
  });

  afterEach(() => {
    localStorage.clear();
  });

  it("should render without crashing", () => {
    localStorage.clear();
    const wrapper = shallow(<EditButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render without crashing when logged in", () => {
    const props = {
      slug: "article-slug",
      hidden: "true",
    };
    const wrapper = shallow(<EditButton {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should map state to props", () => {
    const state = {
      articleCreateReducer: {
      },
    };
    expect(mapStateToProps(state)).toEqual({
      EditState: {},
    });
  });
});
