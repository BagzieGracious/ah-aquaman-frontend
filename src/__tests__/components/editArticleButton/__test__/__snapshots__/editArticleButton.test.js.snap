// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit button component test should render without crashing 1`] = `ShallowWrapper {}`;

exports[`Edit button component test should render without crashing when logged in 1`] = `ShallowWrapper {}`;
